cfme/services/catalogs/catalog_item.py

```diff
@@ -112,23 +197,11 @@ def __init__(self, item_type=None, vm_name=None, name=None, description=None,
         self.provider_type = provider_type
         self.provisioning_data = prov_data
         self.domain = domain
-        Navigatable.__init__(self, appliance=appliance)
-
-    def __str__(self):
-        return self.name
 
     def create(self):
-        # Create has sequential forms, the first is only the provider type
-        navigate_to(self, 'Add')
-        # For element not found exception (To be removed)
-        sel.sleep(5)
+        view = navigate_to(self, 'Add')
         sel.select("//select[@id='st_prov_type']",
                    self.provider_type or self.item_type or 'Generic')
```

Comment:
So I'm not quite following your comment - you would not be calling View.fill on the select field.

In the form/view class you define a before_fill() method that is executed before the fill that you're already calling below.   In that before_fill method you can define a BootstrapSelect widget for the provider type selection and do this selection there.

One of the biggest reasons for this conversion is abstracting selenium out of our page models.  At the very least this needs a #TODO and issue to be refactored more to remove selenium.